src/ValueObject/EncryptedColumn.php

```diff
@@ -2,30 +2,50 @@
 
 namespace Carnage\EncryptedColumn\ValueObject;
 
+use Carnage\EncryptedColumn\Encryptor\HaliteEncryptor;
+use Carnage\EncryptedColumn\Serializer\PhpSerializer;
+
 class EncryptedColumn implements \JsonSerializable
 {
     private $classname;
     private $data;
+    private $encryptor;
+    private $serializer;
 
     /**
      * EncryptedColumn constructor.
      * @param $classname
      * @param $data
      */
-    public function __construct($classname, $data)
-    {
+    public function __construct(
+        $classname,
+        $data,
+        $encryptor = HaliteEncryptor::IDENTITY,
+        $serializer = PhpSerializer::IDENTITY
+    ) {
         $this->classname = $classname;
         $this->data = $data;
+        $this->encryptor = $encryptor;
+        $this->serializer = $serializer;
     }
 
     public static function fromArray(array $data)
     {
-        return new self($data['classname'], $data['data']);
+        if(!isset($data['serializer'])) {
```

Comment:
BC with existing stored data which might not have these fields set; thou there aren't many installs yet so I could just break BC in a 0.x release anyway